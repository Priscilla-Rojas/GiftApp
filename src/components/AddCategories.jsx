import style from "../styles/AddCategories.module.css"
import { useState } from 'react';
import PropTypes from "prop-types"


export const AddCategories = ({ onNewCategory }) => {
  const [inputValue, setimputValue] = useState('')
  const onInputChange = (event)=>{
    setimputValue(event.target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return
    onNewCategory( inputValue.trim());
    setimputValue('');
  }
  return (
    <form onSubmit={ onSubmit } className= {style.form} aria-label="form">
        <input 
          type="text" 
          placeholder="Search gif" 
          value={inputValue} 
          onChange={ onInputChange }
        />
    </form>
  )
}

AddCategories.propTypes={
  onNewCategory: PropTypes.func.isRequired,
}
