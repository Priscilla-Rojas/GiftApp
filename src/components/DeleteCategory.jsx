import style from '../styles/DeleteCategory.module.css'

export const DeleteCategory = ({ onDeleteCategory, category }) => {
  return (
    <>
      <button className={ style.close} onClick={ () => onDeleteCategory(category)} > x </button>
    </>
  )
}
