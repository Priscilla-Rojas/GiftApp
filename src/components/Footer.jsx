import style from '../styles/footer.module.css'

export const Footer = () => {
  return (
    <footer className={ style.footer}>
      <h2>React + Vite</h2>
      <h2>Developed by <a href="https://www.linkedin.com/in/priscilla-k-rojas/" target='_blank' rel='noreferrer'>Priscilla Rojas</a></h2>
    </footer>
  )
}
