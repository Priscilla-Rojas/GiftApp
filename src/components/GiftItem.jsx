import style from "../styles/GiftItem.module.css"

export const GiftItem = ({ title, url }) => {
  
  return (
    <div className={ style.card}>
      <img src={ url } alt={ title }/>
      <p>{ title }</p>
    </div>
  )
}
