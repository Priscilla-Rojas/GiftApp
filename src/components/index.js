
export * from './AddCategories';
export * from './GiftGrid';
export * from './GiftItem';
export * from './Footer';
