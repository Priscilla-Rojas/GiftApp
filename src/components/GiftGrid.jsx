import { GiftItem } from "./GiftItem";
import { useFetchGifts } from "../hooks/useFetchGifts";
import { DeleteCategory } from "./DeleteCategory";

import style from "../styles/GiftGrid.module.css"


export const GiftGrid = ({ category, onDeleteCategory}) => {
  
  const { images, isLoading } = useFetchGifts( category); 
  
  
  return (
    <div className={ style.grid}>
      <section className={ style.header} >
        <h2 className={ style.title }>{ category }</h2>
        <DeleteCategory onDeleteCategory={ onDeleteCategory } category= { category }/>
      </section>
      {
        isLoading && ( <h2> cargando...</h2>) 
      }
      <div className={ style.cards}>
      {images.map( ( image )=>(
        <GiftItem key={ image.id } { ...image } />
      ))}
      </div>
    </div>
    
  )
}
