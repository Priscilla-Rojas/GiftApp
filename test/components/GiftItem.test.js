/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GiftItem"


describe('Pruebas en el Componente GigtItem', () => {
  const title= 'Saitama';
  const url= 'https://one-pounch.com/saitama.jpg';

  test('Debe coincidir con el snapshop', () => { 

    const { container } = render( <GiftItem title={title} url={url}/>);
    expect ( container ).toMatchSnapshot();
  })
  test('Debe mostrar la imagen con el url y el alt indicado', () => { 
    render( <GiftItem title={title} url={url}/>);
    // screen.debug();
    const { src, alt} = screen.getByRole( 'img');
    expect( src ).toBe( url);
    expect( alt ).toBe( alt);
  })
  test('debe de mostrar el titulo en el componente', () => { 

    render( <GiftItem title={title} url={url}/>);
    expect( screen.getByText( title)).toBeTruthy()
  })
})