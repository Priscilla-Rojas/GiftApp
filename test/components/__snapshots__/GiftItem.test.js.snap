// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente GigtItem Debe coincidir con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-pounch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
